rust/main/chains/hyperlane-sealevel/src/provider.rs

```diff
@@ -35,27 +36,14 @@ lazy_static! {
 }
 
 /// A wrapper around a Sealevel provider to get generic blockchain information.
-#[derive(Debug)]
+#[derive(Debug, new)]
 pub struct SealevelProvider {
-    domain: HyperlaneDomain,
     rpc_client: Arc<SealevelRpcClient>,
+    domain: HyperlaneDomain,
     native_token: NativeToken,
 }
 
 impl SealevelProvider {
-    /// Create a new Sealevel provider.
-    pub fn new(domain: HyperlaneDomain, conf: &ConnectionConf) -> Self {
-        // Set the `processed` commitment at rpc level
-        let rpc_client = Arc::new(SealevelRpcClient::new(conf.url.to_string()));
-        let native_token = conf.native_token.clone();
-
-        Self {
-            domain,
-            rpc_client,
-            native_token,
-        }
-    }
```

Comment:
my preference is to keep `conf: &ConnectionConf` as a constructor argument so `native_token` can be read from it, rather than have the token passed explicitly - which feels like it'd break some encapsulation

also since `SealevelRpcClient` is not generic we can also instantiate it inside the constructor rather than have it passed as an argument, to further encapsulate

Reply:
I can update it to keep `conf: &ConnectionConf`, but for `SealevelRpcClient` I'm not sure what you mean?
I pass it in this way so we can initialize `RpcClient` and `SealevelRpcClient` with metrics outside the constructor, so we don't need to pass metrics around everywhere.

Reply:
We also need to pass in `Option<ChainInfo>`